website/content/docs/k8s/installation/vault/enterprise-license.mdx

```diff
@@ -0,0 +1,98 @@
+---
+layout: docs
+page_title: Storing Enterprise License in Vault
+description: >-
+  Configuring the Consul Helm chart to use enterprise license stored in Vault.
+---
+
+# Storing the Enterprise License in Vault
+
+To use an enterprise license stored in Vault, the steps will be similar to [Storing Gossip Encryption Key in Vault](/docs/k8s/installation/vault/gossip). You need to do the following:
+
+1. Store an enterprise license key in Vault.
+1. Create policies that will allow Consul client and server to access that key.
+1. Create a Kubernetes auth roles that link policies from step 2 to Kubernetes service accounts of the Consul servers and clients.
+
+## Configuring Vault
+
+First, store the license key in Vault:
+
+```shell-session
+$ vault kv put secret/consul/enterpriselicense key="<enterprise license>"
+```
+
+Next, you will need to create a policy that allows read access to this secret:
+
+
+<CodeBlockConfig filename="enterpriselicense-policy.hcl">
+
+```HCL
+path "secret/data/consul/enterpriselicense" {
+  capabilities = ["read"]
+}
+```
+
+</CodeBlockConfig>
+
+```shell-session
+$ vault policy write enterpriselicense-policy enterpriselicense-policy.hcl
+```
+
+Prior to creating Vault auth roles for the Consul servers and clients, ensure that the Vault Kubernetes auth method is enabled as described in [Vault Kubernetes Auth Method](/docs/k8s/installation/vault#vault-kubernetes-auth-method).
+
+Next, you will create Kubernetes auth roles for the Consul server and client:
+
+```shell-session
+$ vault write auth/kubernetes/role/consul-server \
+    bound_service_account_names=<Consul server service account> \
+    bound_service_account_namespaces=<Consul installation namespace> \
+    policies=enterpriselicense-policy \
+    ttl=1h
+```
+
+```shell-session
+$ vault write auth/kubernetes/role/consul-client \
+    bound_service_account_names=<Consul client service account> \
+    bound_service_account_namespaces=<Consul installation namespace> \
+    policies=enterpriselicense-policy \
+    ttl=1h
+```
+
+To find out the service account names of the Consul server and client,
+you can run the following `helm template` commands with your Consul on Kubernetes values file:
+
+- Generate Consul server service account name
+  ```shell-session
+  $ helm template --release-name ${RELEASE_NAME} -s templates/server-serviceaccount.yaml hashicorp/consul
+  ```
+
+- Generate Consul client service account name
+  ```shell-session
+  $ helm template --release-name ${RELEASE_NAME} -s templates/client-serviceaccount.yaml hashicorp/consul
+  ```
+
+## Deploying the Consul Helm chart
+
+Now that you have configured Vault, you can configure the Consul Helm chart to
+use the enterprise license key in Vault:
+
+<CodeBlockConfig filename="values.yaml">
+
+```yaml
+global:
```

Comment:
I'm wondering if there is a way to phrase that the below is a snippet of a values.yaml and not a valid values.yaml so that someone does not follow this guide word-for-word and end up stuck?
Specifically it is missing most of the required configuration for vault as well as stuff like server/clients being enabled.
IMHO it would be safe to just say "these modifications are required on top of a normal configuration" or something like this.